indra/newview/llviewerjointattachment.h

```diff
@@ -98,16 +98,40 @@ class LLViewerJointAttachment :
 	typedef std::vector<LLPointer<LLViewerObject> > attachedobjs_vec_t;
 	attachedobjs_vec_t mAttachedObjects;
 
+    bool    hasChanged(const LLVector3 &pos, const LLQuaternion &rot) const
+    {
+        static constexpr F32 SMALL_CHANGE(0.05 * 0.05);
+
+        LLVector3 unit_vec(LLVector3::x_axis * rot);
+        F32 unit_cos(0);
+        if (!mLastTrackedUnit.isNull())
+        {
+            unit_cos = unit_vec * mLastTrackedUnit;
+        }
+        mLastTrackedUnit = unit_vec;
+
+        return (((pos - mLastTrackedPos).magVecSquared() > SMALL_CHANGE) || (unit_cos < 0.9));
+    }
```

Comment:
It looks like you're using `mLastTrackedUnit` to figure out how much the rotation has changed but there is a better way to do it:

`LLQuaternion::almost_equal(rot, mLastTrackedRot, minimum_angle)`.

Reply:
Nice.  I'll switch to that.